File: java_projects/glue_code/commons-graph/src/main/java/org/apache/commons/graph/builder/DefaultGrapher.java
package org.apache.commons.graph.builder;

import org.apache.commons.graph.ContextInitializer;
import org.apache.commons.graph.MutableGraph;
import org.graalvm.polyglot.Value;

final class DefaultGrapher<V, E> implements GraphConnector<V, E> {
  private static Value clz =
      ContextInitializer.getPythonClass("/builder/DefaultGrapher.py", "DefaultGrapher");
  private Value obj;

  public DefaultGrapher(Value obj) {
    this.obj = obj;
  }

  public Value getPythonObject() {
    return obj;
  }

  public <N extends V> N addVertex(N node) {
    //
    // node = checkNotNull(node, "Null vertex not admitted");
    // graph.addVertex(node);
    // return node;
    //

    // TODO: Check the type mapping below!
    return obj.invokeMember("addVertex", node).as(N.class);
  }

  public <A extends E> HeadVertexConnector<V, E> addEdge(A arc) {
    //
    // arc = checkNotNull(arc, "Null edge not admitted");
    // return new DefaultHeadVertexConnector<V, E>(graph, arc);
    //

    // TODO: Check the type mapping below!
    return obj.invokeMember("addEdge", arc).as(HeadVertexConnector.class);
  }

  public DefaultGrapher(MutableGraph<V, E> graph) {
    //
    // this.graph = graph;
    //

    this.obj = clz.invokeMember("__init__", graph);
  }
}

File: data/recomposed_projects/deepseek-coder-33b-instruct/commons-graph/src/main/org/apache/commons/graph/builder/DefaultGrapher.py
from __future__ import annotations
import io
import typing
from typing import *
from src.main.org.apache.commons.graph.MutableGraph import *
from src.main.org.apache.commons.graph.builder.DefaultHeadVertexConnector import *
from src.main.org.apache.commons.graph.builder.GraphConnector import *
from src.main.org.apache.commons.graph.builder.HeadVertexConnector import *
from src.main.org.apache.commons.graph.utils.Assertions import *


class DefaultGrapher(GraphConnector):

    __graph: MutableGraph[typing.Any, typing.Any] = None

    def addVertex(self, node: typing.Any) -> typing.Any:

        # Calling the checkNotNull method from Assertions class
        node = Assertions.checkNotNull(node, "Null vertex not admitted")

        # Adding the vertex to the graph
        self.__graph.addVertex(node)

        return node

    def addEdge(self, arc: typing.Any) -> HeadVertexConnector[typing.Any, typing.Any]:

        # Calling the checkNotNull method from Assertions class
        arc = Assertions.checkNotNull(arc, "Null edge not admitted")

        return DefaultHeadVertexConnector(self.__graph, arc)

    def __init__(self, graph: MutableGraph[typing.Any, typing.Any]) -> None:

        self.__graph = graph
